app/utils/eosio.js

```diff
@@ -334,19 +329,70 @@ class EosioService {
     chainId: process.env.SCATTER_CHAINID || this.node.chainID,
   });
 
-  getNode = () => {
-    const node = JSON.parse(localStorage.getItem(LOCAL_STORAGE_BESTNODE));
-    return node || getBestNode();
+  handleCaseWithInvalidNode = async errorMsg => {
+    try {
+      if (!this.isInvalidNodeHandling) {
+        this.isInvalidNodeHandling = true;
+
+        const storedNodes = this.getStoredNodes();
+
+        if (!storedNodes || !storedNodes.length) throw new Error(errorMsg);
+
+        const freshNodes = storedNodes.filter((_, index) => index !== 0);
+        localStorage.setItem(ENDPOINTS_LIST, JSON.stringify(freshNodes));
+
+        if (this.eosApi.signatureProvider) {
```

Comment:
it is better to add boolean property `isScatter` to eosApi to make code easier to read.